connector/dialect.tdd

```diff
@@ -380,60 +151,66 @@
       <argument type='datetime' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>(CASE
-	WHEN %1 THEN 1.0
-	WHEN NOT %1 THEN 0.0
-	ELSE NULL END)</formula>
-      <argument type='bool' />
+      <formula>CAST(%1 AS DECIMAL)</formula>
```

Comment:
I see this function is named "FLOAT" but we're casting the argument as a `DECIMAL`. Is that the appropriate type mapping? I thought "FLOAT" more closely maps to the MongoSQL `DOUBLE` type. MongoSQL does support casting to `DECIMAL`, I'm just surprised by the name.

Reply:
I added a `DECIMAL` cast function and changed `FLOAT` to cast to `DOUBLE`.